api.go

```diff
@@ -258,93 +304,92 @@ func RecoverCluster(conf *Config, fsm FSM, logs LogStore, stable StableStore,
 	// expect data to be there and it's not. By refusing, we force them
 	// to show intent to start a cluster fresh by explicitly doing a
 	// bootstrap, rather than quietly fire up a fresh cluster here.
-	hasState, err := HasExistingState(logs, stable, snaps)
+	hasState, err = HasExistingState(logs, stable, snaps)
 	if err != nil {
 		return fmt.Errorf("failed to check for existing state: %v", err)
 	}
 	if !hasState {
 		return fmt.Errorf("refused to recover cluster with no initial state, this is probably an operator error")
 	}
+	return nil
+}
+
+// restoreSnapshots attempts to restore each snapshot
+func restoreSnapshots(snaps SnapshotStore, fsm FSM) (index, term uint64, err error) {
+	var snapshots []*SnapshotMeta
+	snapshots, err = snaps.List()
 
-	// Attempt to restore any snapshots we find, newest to oldest.
-	var snapshotIndex uint64
-	var snapshotTerm uint64
-	snapshots, err := snaps.List()
 	if err != nil {
-		return fmt.Errorf("failed to list snapshots: %v", err)
+		return 0, 0, fmt.Errorf("failed to list snapshots: %v", err)
 	}
+
 	for _, snapshot := range snapshots {
-		_, source, err := snaps.Open(snapshot.ID)
+		var source io.ReadCloser
```

Comment:
I prefer explicit style just because it is cleaner to me, but I don't think there is a set pattern throughout the lib (I see some times it being explicit here, while others it is not)
I was trying to do it the most idiomatic way, so if `:=` is more idiomatic I'd be happy to switch it :)